lib/ui/profitloss/RecordAmountHeader.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:farmsmart_flutter/ui/common/RegExInputFormatter.dart';
+import 'package:farmsmart_flutter/ui/profitloss/RecordAmount.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class RecordAmountHeaderViewModel {
+  String amount;
+  final Function(String) listener;
+  bool isEditable;
+
+  RecordAmountHeaderViewModel({this.amount, this.listener, this.isEditable});
+}
+
+class RecordAmountHeaderStyle {
+  final TextStyle hintTextStyle;
+  final TextStyle titleTextStyle;
+
+  final EdgeInsets edgePadding;
+
+  final double height;
+  final int maxLines;
+
+  RecordAmountHeaderStyle(
+      {this.hintTextStyle,
+      this.titleTextStyle,
+      this.edgePadding,
+      this.height,
+      this.maxLines});
+
+  RecordAmountHeaderStyle copyWith(
+      {TextStyle hintTextStyle,
+      TextStyle titleTextStyle,
+      EdgeInsets edgePadding,
+      double height,
+      int maxLines}) {
+    return RecordAmountHeaderStyle(
+        hintTextStyle: hintTextStyle ?? this.hintTextStyle,
+        titleTextStyle: titleTextStyle ?? this.titleTextStyle);
+  }
+
+  factory RecordAmountHeaderStyle.defaultCostStyle() {
+    return RecordAmountHeaderStyle(
+        hintTextStyle: const TextStyle(
+            fontSize: 72,
+            fontWeight: FontWeight.w500,
+            color: Color(0x4cff8d4f), letterSpacing: 4.32),
+        titleTextStyle: const TextStyle(
+            fontSize: 72,
+            fontWeight: FontWeight.w500,
+            color: Color(0xFFff8d4f), letterSpacing: 4.32));
+  }
+
+  factory RecordAmountHeaderStyle.defaultSaleStyle() {
+    return RecordAmountHeaderStyle.defaultCostStyle().copyWith(
+        hintTextStyle: const TextStyle(
+            fontSize: 72,
+            fontWeight: FontWeight.w500,
+            color: Color(0x4c24d900), letterSpacing: 4.32),
+        titleTextStyle: const TextStyle(
+            fontSize: 72,
+            fontWeight: FontWeight.w500,
+            color: Color(0xFF24d900),letterSpacing: 4.32),
+        edgePadding: const EdgeInsets.symmetric(horizontal: 32),
+        height: 137.5,
+        maxLines: 1);
+  }
+}
+
+class RecordAmountHeader extends StatelessWidget {
+  final RecordAmountHeaderViewModel _viewModel;
+  final RecordAmountHeaderStyle _style;
+
+  RecordAmountHeader(
+      {Key key,
+      RecordAmountHeaderViewModel viewModel,
+      RecordAmountHeaderStyle style})
+      : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final _amountValidator = RegExInputFormatter.withRegex(
+        '^\$|^(0|([1-9][0-9]{0,3}))(\\.[0-9]{0,2})?\$');
```

Comment:
no inline string literals like this, they must be in a private _Strings. or _Constants at the top of the file.

Reply:
Done